### ISSUE TITLE
No release version control

### ISSUE THREAD

**Issue by Ilyesbdlala** [whoot/Typo3Scan#19]
**Is your feature request related to a problem? Please describe.**\
Typo3Scan only makes the latest version of the software available. As a Typo3Scan user i would like to know and decide which version of Typo3Scan to run in my infrastructure, and when to update. 
 
**Describe the solution you'd like**\
It would be nice to have [github releases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) that keep track when a new version of Typo3Scan comes out. ([Example](https://github.com/returntocorp/semgrep/releases))

**Describe alternatives you've considered**\
Tagging the branch of each release is also an alternative. As it would allow the user to choose which version to check out. ([Example](https://github.com/Tuhinshubhra/CMSeeK/tags))

**Comment by whoot** [comment]
![but-why-wtf](https://user-images.githubusercontent.com/8066322/143131132-2318fa52-bc86-419a-9ecf-d32e35553239.gif)

Jokes aside. Maybe you can elaborate more on why this is needed.

At the current state Typo3Scan only receives database updates by meaning hashes of new Typo3 releases and a list of available extensions. Not updating it would simply result in not being able to identify (some) Typo3 versions and extensions. This would mean that I would have to publish a new release with every database update which is a bit of an overkill imho.
Any code changes to Typo3Scan itself are tracked in the Changelog.

**Comment by Ilyesbdlala** [comment]
We use Typo3Scan as a security scanner in the open-source project [secureCodeBox](https://github.com/secureCodeBox/secureCodeBox). It's basically a Kubernetes based, modularized tool-chain for continuous security scans. We build the scanners as docker images. We add features to the project with each release , upgrade the scanners to newer versions and we make sure that every scanner works through Integration tests. 
We avoid automatic changes to not allow any breaking changes to the scanners. We prefer to let our GitHub bot detect new versions and open a PR for it. We then review the changes and apply them. So easily knowing which version we're on and which we're switching to is important to us.
Long story short, I think it's enough for me to consider this issue resolved is if you tag the current version of Typo3Scan, and apply any non-breaking changes to it. A version with breaking changes then receives a new version number. 
A breaking change for us, is any change that changes the JSON output Schema. That can be different to other people.
Hopefully this is something you're willing to do, considering you're doing this in your free time :) 

PS: The GIF above repeatedly saying to me "But Why?" the whole time I'm writing this, was pretty amusing ^_^

**Comment by definity** [comment]
Well.. there is versioning in the tool itself, it's just not reflected in Github.
https://github.com/whoot/Typo3Scan/blob/master/typo3scan.py#L21
```python
__version__ = '0.7.3'
```

Is that all you want to know?  In other words, when the version number changes you would trigger an automation that would rebuild your container.

Could @whoot maybe just add tags whenever you increment the version number?  That _should_ capture changes made that don't affect the version database and more easily fit in with your current way of working 🤷🏻‍♂️

**Comment by Ilyesbdlala** [comment]
> Well.. there is versioning in the tool itself, it's just not reflected in Github. https://github.com/whoot/Typo3Scan/blob/master/typo3scan.py#L21
> 
> ```python
> __version__ = '0.7.3'
> ```
> 
> Is that all you want to know? In other words, when the version number changes you would trigger an automation that would rebuild your container.
> 
> Could @whoot maybe just add tags whenever you increment the version number? That _should_ capture changes made that don't affect the version database and more easily fit in with your current way of working 🤷🏻‍♂️

Yes exactly. It would be nice to have the version number reachable by the GitHub API

**Comment by definity** [comment]
Yeah, the alternative is following the repo and after enough changes have been made update containers (we're doing something similar to what you're doing but in GCP with Cloud Run).  Not elegant but it works.  Ideally we'd do more of what you're doing and build/deploy images automatically when there's a new release.

**Comment by whoot** [comment]
Got it. Thanks for clarification.

Actually, I perform version jumps only after code changes, but not after a database update.
Currently there is only one topic on my todo list for Typo3Scan: abolish the config file.
I´ve already started with this months ago, but I simply do not have the time to finish it, because it requires a lot of code refactoring. Hopefully I get this done during my Christmas vacation. After that the tool is "done" and I would bump the version to "1.0 final" or something like that and create a github release.
Hopefully I just need to keep the database up to date afterwards.


PS: I am happy you liked the gif :D

**Comment by whoot** [comment]
I finally had the time to work on Typo3Scan again and proudly can announce the release of version 1.0 🎉

**Comment by Ilyesbdlala** [comment]
> I finally had the time to work on Typo3Scan again and proudly can announce the release of version 1.0 tada

Congratz :)
